### DISCUSSION TITLE
Storefront Variant Metafield Filtering seems to have stopped working

### DISCUSSION THREAD

**Discussion by Mattgh91** [Shopify/storefront-api-feedback#190] [Help] [answered]
Hi,

I have a site set up which was reliant on filtering products using the variant Metafields as such:

```
collection(handle: "${categoryHandle}") {
            handle            
            products(first: 24,            
                filters: {                
                    variantMetafield: {                    
                        namespace: "custom",                        
                        key: "skin_tone",                        
                        value: "${skintone}"                        
                    }                    
                }                
            )
```

which used to work just fine and return the expected array of product nodes, but as of a few month ago the amount of API versions which appeared to support this seemed to narrow, and now all API versions are returning an empty array when filtering products this way.

Has something changed in how we should be calling products via this filtering method, or has the functionality been broken?

Thank you,
Matt.

**Comment by johncraigcole** [comment]
Hey @Mattgh91 👋  I haven't had any problems in running some tests today using a variant metafield. 

I know you're likely already aware, but the collection variant metafield filters are dependent on the filters being  defined in the Search & Discovery App, which can be found under `Online Store -> Navigation -> Collection and search filters`. Just wanted to make sure everything is setup ok there.

Another possible issue could be the collection or the products contained within it are no longer published to the app channel being used to make the Storefront API request.

Could you also share the the api version you're using to make the requests. If I'm unable to replicate, it might be best to go through Shopify support channels, so you get authenticated on the store in question and support can access the store to hopefully gain context on what might be happening.

**Comment by Mattgh91** [comment]
Hi @johncraigcole, thanks for the reply.

I do think that it looks to be set up correctly in the Collection and search filters channel:
![Screenshot 2023-07-03 at 16 04 44](https://github.com/Shopify/storefront-api-feedback/assets/15674732/bb763ec3-9166-461d-8d59-16b03b11b043)

and is referencing the variant metafield:
![Screenshot 2023-07-03 at 16 05 36](https://github.com/Shopify/storefront-api-feedback/assets/15674732/b2d77e81-f11c-401d-8100-683a38fd05d1)

In fact, these filters are still working on the Front End when they're consumed by Liquid, and not by the Storefront API, as I can [demo here](https://minifigs.me/collections/all-heads-new-1?filter.v.m.custom.skin_tone=Light).

![minifigs-dev myshopify com_admin_settings_apps_development_6708756481_configuration_storefront_api_integration (2)](https://github.com/Shopify/storefront-api-feedback/assets/15674732/03312ac4-1782-4869-ba1b-3793a4ef5320)
![minifigs-dev myshopify com_admin_settings_apps_development_6708756481_configuration_storefront_api_integration (3)](https://github.com/Shopify/storefront-api-feedback/assets/15674732/44965ece-726e-4ca6-8b17-69ec177cbd93)
- Picture of current Storefront permissions


This was working fine on several versions of the Storefront API including 2022-07, 2023-01, 2023-04, and the unstable release at the time, but one by one the feature seemed to be slowly killed off (even on then-live API versions).

I appreciate it if you can offer any insight at all on this, even if you can provide a demo as to what your API calls might look like if there are any differences from mine, as I have been in touch with Shopify support extensively on this but we don't appear to be getting anywhere unfortunately.

**Comment by johncraigcole** [comment]
Thanks @Mattgh91 for the additional context. In your tests, how big are the collections you're filtering against? Also, could you share the most recent support ticket number?

**Comment by Mattgh91** [comment]
No problem!

The collections are generally 100-200 products in size.

I believe we may have had a couple of tickets over the course of it now, but the original ticket is 35527297.

**Comment by johncraigcole** [comment]
Hey @Mattgh91 had a peak at the ticket. Would you mind making a fresh request that isn't working and share the response header `x-request-id` for the request that returns no result and I'll take a look.

**Reply by johncraigcole** [reply, marked as answer]
The other thing to check is just make sure your `skin_tone` filter is a case sensitive match to the variant metafield value.

**Comment by Mattgh91** [comment]
HI @johncraigcole - yep of course!

Fresh request from just now on an unpublished theme: `x-request-id: 573a2204-1215-46f8-87c1-de57a2e40196`

Yep, the cases are correct, unfortunately.

Thanks again for your help!

**Reply by Mattgh91** [reply]
You should now be able to see a live demo of this actually if that would help, but using another product & collection which uses variant metafield filtering as the heads are too prominent to have broken.

If you navigate to https://minifigs.me/pages/create-your-own, and then enter a name, select a skin tone, and then click on "Legs" -> "Colour" -> Click any of the colour filters, then you should be able to see the request in full there.

**Reply by johncraigcole** [reply]
Hey @Mattgh91 thanks for the request id. Since we're getting a little into merchant specific info I'm going to reply on a follow up ticket. You should see a reply from me on ticket # 39729758.
